app/portal/admin/page.js

```diff
@@ -65,6 +66,44 @@ const demoData = [
   { id: 5, name: "Test", quantity: 0, showButton: true },
 ]
 
+const showDataTable = (data) => {
+  const tableLength = Object.keys(data).length
```

Comment:
This is a really smooth way of checking to see if there's any data to render, nice thinking here!